instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -227,10 +232,6 @@ def client_response_hook(span: Span, message: dict):
     remove_url_credentials,
 )
 
-_ServerRequestHookT = typing.Optional[typing.Callable[[Span, dict], None]]
-_ClientRequestHookT = typing.Optional[typing.Callable[[Span, dict], None]]
-_ClientResponseHookT = typing.Optional[typing.Callable[[Span, dict], None]]
```

Comment:
Extracted these into `types.py` and made public, so that they could be reused in the FastAPI and Starlette instrumentations